File: Bind75/2390/Solution.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    let res  =[];
    for(let i =0;i<s.length;i++){
        let c = s[i];
        if (c ==="*"){
            if(res.length>0){
                res.pop();            }
        }
        else{
            res.push(c)
        }
    }
    return res.join("");

};

File: Bind75/2390/Solution.java
import java.util.ArrayList;

class Solution {
    public String removeStars(String s) {
        ArrayList<Character> res = new ArrayList<>();
        
        for(int i =0;i<s.length();i++){
            char c = s.charAt(i);
            if (c =='*'){
                if(!res.isEmpty()){
                    res.remove(res.size()-1);
                }
            }else{
                res.add(c);
            }
        }
        
        StringBuilder r = new StringBuilder();
        for(char c :res){
            r.append(c);
        }
        return r.toString();
    }
}